src/python/pants/engine/scheduler.py

```diff
@@ -132,45 +128,13 @@ def __init__(
 
     def graph_trace(self, session, execution_request):
         with temporary_file_path() as path:
-            self._native.lib.graph_trace(self._scheduler, session, execution_request, path.encode())
+            self._native.lib.graph_trace(self._scheduler, session, execution_request, path)
```

Comment:
Huh, so it's safe to serialize unicode? Neat!

Reply:
It's not inherently zero copy (some conversion still happens at the call boundary), but it is possible to do some things as zero copy by accepting a `PyString` on the rust side, and then temporarily viewing it as a `Cow<str>` (copy on write `&str`): https://docs.rs/cpython/0.5.0/cpython/struct.PyString.html#method.to_string